analyzers/tests/SonarAnalyzer.UnitTest/TestCases/GetTypeWithIsAssignableFrom.cs

```diff
@@ -92,6 +92,33 @@ public void Go(Repro_3605 value)
         }
     }
 
+    // https://github.com/SonarSource/sonar-dotnet/issues/6616
+    public class Repro_6616
+    {
+        public void IsInstanceOfType(object obj)
+        {
+            _ = typeof(ISet<int>).IsInstanceOfType(obj); // Noncompliant, bounded generic type
+            _ = typeof(ISet<>).IsInstanceOfType(obj); // Compliant, unbonded generic type
+            _ = typeof(IDictionary<int, int>).IsInstanceOfType(obj); // Noncompliant, bounded generic type
+            _ = typeof(IDictionary<,>).IsInstanceOfType(obj); // Compliant, unbonded generic type
+
+            Type t = typeof(ISet<>);
+            _ = t.IsInstanceOfType(obj); // Compliant, not a typeof expression
+        }
+
+        public void IsAssignableFrom(object obj)
+        {
+            _ = typeof(HashSet<int>).IsAssignableFrom(obj.GetType()); // Noncompliant, bounded generic type
+            _ = typeof(HashSet<>).IsAssignableFrom(obj.GetType()); // Compliant, unbonded generic type
+            _ = typeof(Dictionary<int, int>).IsAssignableFrom(obj.GetType()); // Noncompliant, bounded generic type
+            _ = typeof(Dictionary<,>).IsAssignableFrom(obj.GetType()); // Compliant, unbonded generic type
+
+            Type t1 = typeof(ISet<>);
+            Type t2 = obj.GetType();
```

Comment:
Same question as above

Reply:
Actually having some other usage of `typeof(ISet<>)` is handy, so it's probably fine to have it here

Reply:
I could apply the same suggestion as in https://github.com/SonarSource/sonar-dotnet/pull/6675#discussion_r1091936330